src/overlays/actors/ovl_Boss_Va/z_boss_va.c

```diff
@@ -173,157 +293,3723 @@ static ColliderQuadInit D_8095BFF4 = {
+                    sp80.y -= ptr->velocity.y + 4.0f;
+                    ptr->rot.x += 0x1770;
+                    floorY = BgCheck_EntityRaycastFloor1(&globalCtx->colCtx, &sp7C, &sp80);
+                    if ((sp7C != NULL) && (ptr->pos.y <= floorY)) {
+                        ptr->mode = GORE_FLOOR;
+                        ptr->timer = 30;
+                        ptr->pos.y = floorY + 1.0f;
+                        ptr->accel = ptr->velocity = sZeroVec;
+                        ptr->rot.x = -0x4000;
+                    }
+
+                    if (!ptr->timer) {
+                        ptr->type = VA_NONE;
+                    }
+
+                } else if (ptr->mode == GORE_FADING) {
+                    if (ptr->timer == 0) {
+                        ptr->type = VA_NONE;
+                        if (1) {}
+                    }
+
+                } else {
+                    Math_SmoothStepToF(&ptr->scaleMod, 0.075f, 1.0f, 0.005f, 0.0f);
+                    Math_SmoothStepToF(&ptr->vaGorePulseRate, 0.0f, 0.6f, 0.005f, 0.0013f);
+                    if ((globalCtx->gameplayFrames % 4) == 0) {
+                        Math_SmoothStepToS(&ptr->primColor[0], 95, 1, 1, 0);
+                    }
+                }
+                ptr->vaGorePulse += ptr->vaGorePulseRate;
+            }
+
+            if (ptr->type == VA_TUMOR) {
+                refActor = ptr->parent;
+
+                ptr->rot.z += 0x157C;
+                ptr->envColor[3] = (s16)(Math_SinS(ptr->rot.z) * 50.0f) + 80;
+                Math_SmoothStepToF(&ptr->scale, ptr->scaleMod, 1.0f, 0.01f, 0.005f);
+                ptr->pos.x = ptr->offset.x + refActor->actor.world.pos.x;
+                ptr->pos.y = ptr->offset.y + refActor->actor.world.pos.y;
+                ptr->pos.z = ptr->offset.z + refActor->actor.world.pos.z;
+
+                switch (ptr->mode) {
+                    case TUMOR_UNUSED:
+                        if (ptr->timer == 0) {
+                            yaw = Math_Vec3f_Yaw(&refActor->actor.world.pos, &ptr->pos);
+                            ptr->type = VA_NONE;
+                            BossVa_BloodSplatter(globalCtx, ptr, yaw, ptr->scale * 4500.0f, 1);
+                            BossVa_Gore(globalCtx, ptr, yaw, ptr->scale * 1.2f);
+                        }
+                        break;
+                    case TUMOR_BODY:
+                    case TUMOR_ARM:
+                        if (refActor->burst) {
+                            ptr->type = VA_NONE;
+                            yaw = Math_Vec3f_Yaw(&refActor->actor.world.pos, &ptr->pos);
+                            BossVa_BloodSplatter(globalCtx, ptr, yaw, ptr->scale * 4500.0f, 1);
+                            BossVa_Gore(globalCtx, ptr, yaw, ptr->scale * 1.2f);
+                        }
+                        break;
+                }
+                if (1) {}
+            }
+        }
+    }
+}
+
+void BossVa_DrawEffects(BossVaEffect* ptr, GlobalContext* globalCtx) {
```

Comment:
```suggestion
void BossVa_DrawEffects(BossVaEffect* effectPtr, GlobalContext* globalCtx) {
```
Or just `effect` and `effectHead`